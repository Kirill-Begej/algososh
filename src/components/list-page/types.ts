import { ElementStates } from "../../types/element-states";

export enum Position {
  Top = "top",
  Bottom = "bottom",
}

export enum Operation {
  AddHead = "addHead",
  AddTail = "addTail",
  DeleteHead = "deleteHead",
  DeleteTail = "deleteTail",
  AddByIndex = "addByIndex",
  DeleteByIndex = "deleteByIndex",
}

export interface ILinkedListNode<T> {
  _value: T;
  next: ILinkedListNode<T> | null;

  get getValue(): T;
  set setValue(value: T);  
}

export interface ILinkedList<T> {
  head: ILinkedListNode<T> | null;
  tail: ILinkedListNode<T> | null;

  toArray(): ILinkedListNode<T>[];
  prepend(value: T): ILinkedList<T>;
  append(value: T): ILinkedList<T>;
  deleteHead(): void;
  deleteTail(): void;
  addByIndex(value: T, index: number): ILinkedList<T>;
  deleteByIndex(vindex: number): ILinkedList<T>;
}

export type TSteps = {
  index: number | null;
  value: string | undefined;
  list: ILinkedListNode<string>[];
  state: ElementStates[];
  direction: Position | null;
  hideValue: boolean[];
}