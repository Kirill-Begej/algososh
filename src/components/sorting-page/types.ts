export type TStep = {
  currentArray: number[];
  elementA: number | null;
  elementB: number | null;
  sortedElements: number[];
};

export enum SortSelection {
  Selection = "selection",
  Bubble = "bubble",
}
