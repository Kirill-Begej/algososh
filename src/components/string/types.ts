import { ElementStates } from "../../types/element-states";

export type TReversingStringResult = {
  steps: string[][];
  state:
    | ElementStates.Default[][]
    | ElementStates.Modified[][]
    | ElementStates.Changing[][];
};
