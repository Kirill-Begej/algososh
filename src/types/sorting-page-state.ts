type TStep = {
  currentArray: number[];
  elementA: number | null;
  elementB: number | null;
  sortedElements: number[]
};

export type TSortingResult = TStep[];