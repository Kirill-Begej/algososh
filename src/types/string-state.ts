import { ElementStates } from "./element-states";

export type TReversingStringResult = {
  steps: string[][];
  state: ElementStates.Default[][] | ElementStates.Modified[][] | ElementStates.Changing[][];
}