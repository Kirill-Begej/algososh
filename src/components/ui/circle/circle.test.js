import renderer from 'react-test-renderer';
import { Circle } from './circle';
import { ElementStates } from '../../../types/element-states';

describe('Тестирование компонента Circle:', () => {
  it('Проверка корректности отрисовки компонента Circle без буквы', () => {
    const circle = renderer.create(<Circle />).toJSON();

    expect(circle).toMatchSnapshot();
  });

  it('Проверка корректности отрисовки компонента Circle с буквами', () => {
    const circle = renderer.create(<Circle letter="ABCD" />).toJSON();

    expect(circle).toMatchSnapshot();
  });

  it('Проверка корректности отрисовки компонента Circle со строкой в head', () => {
    const circle = renderer.create(<Circle head="ABCD" />).toJSON();

    expect(circle).toMatchSnapshot();
  });

  it('Проверка корректности отрисовки компонента Circle с react-элементом в head', () => {
    const circle = renderer.create(<Circle head={<Circle />} />).toJSON();

    expect(circle).toMatchSnapshot();
  });

  it('Проверка корректности отрисовки компонента Circle со строкой в tail', () => {
    const circle = renderer.create(<Circle tail="ABCD" />).toJSON();

    expect(circle).toMatchSnapshot();
  });

  it('Проверка корректности отрисовки компонента Circle с react-элементом в tail', () => {
    const circle = renderer.create(<Circle tail={<Circle />} />).toJSON();

    expect(circle).toMatchSnapshot();
  });

  it('Проверка корректности отрисовки компонента Circle с числом в index', () => {
    const circle = renderer.create(<Circle index="013" />).toJSON();

    expect(circle).toMatchSnapshot();
  });

  it('Проверка корректности отрисовки компонента Circle с isSmall ===  true', () => {
    const circle = renderer.create(<Circle isSmall={true}/>).toJSON();

    expect(circle).toMatchSnapshot();
  });

  it('Проверка корректности отрисовки компонента Circle в состоянии default', () => {
    const circle = renderer.create(<Circle state={ElementStates.Default} />).toJSON();

    expect(circle).toMatchSnapshot();
  });

  it('Проверка корректности отрисовки компонента Circle в состоянии changing', () => {
    const circle = renderer.create(<Circle state={ElementStates.Changing} />).toJSON();

    expect(circle).toMatchSnapshot();
  });

  it('Проверка корректности отрисовки компонента Circle в состоянии modified', () => {
    const circle = renderer.create(<Circle state={ElementStates.Modified} />).toJSON();

    expect(circle).toMatchSnapshot();
  });
});