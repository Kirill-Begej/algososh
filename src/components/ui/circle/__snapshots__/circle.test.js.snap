// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента Circle: Проверка корректности отрисовки компонента Circle без буквы 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle: Проверка корректности отрисовки компонента Circle в состоянии changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle: Проверка корректности отрисовки компонента Circle в состоянии default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle: Проверка корректности отрисовки компонента Circle в состоянии modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle: Проверка корректности отрисовки компонента Circle с isSmall ===  true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle: Проверка корректности отрисовки компонента Circle с react-элементом в head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle: Проверка корректности отрисовки компонента Circle с react-элементом в tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Тестирование компонента Circle: Проверка корректности отрисовки компонента Circle с буквами 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      ABCD
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle: Проверка корректности отрисовки компонента Circle с числом в index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    013
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Тестирование компонента Circle: Проверка корректности отрисовки компонента Circle со строкой в head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    ABCD
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Тестирование компонента Circle: Проверка корректности отрисовки компонента Circle со строкой в tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    ABCD
  </div>
</div>
`;
